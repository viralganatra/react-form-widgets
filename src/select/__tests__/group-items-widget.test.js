import React from 'react';
import GroupItems from '../group-items-widget';

jest.mock('../group-item-widget', () => jest.fn((props) => <li {...props} />));

function noop() {}

const props = {
    items: [
        {
            label: 'test 1',
            value: 1,
        },
        {
            label: 'test 2',
            value: 2,
        },
    ],
    onItemSelection: noop,
};

describe('GroupItemsWidget', () => {
    it('should render an array of items', () => {
        const wrapper = shallow(<GroupItems {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should validate the prop types', () => {
        const wrapperInvalidItems = () => shallow(
            <GroupItems {...props} items={[{ value: 'foo' }]} />,
        );

        const wrapperInvalidOnItemSelection = () => shallow(
            <GroupItems {...props} onItemSelection={2} />,
        );

        expect(wrapperInvalidItems).toThrowErrorMatchingSnapshot();
        expect(wrapperInvalidOnItemSelection).toThrowErrorMatchingSnapshot();
    });

    it('should call the props method onItemSelection with the item as an argument', () => {
        const methodSpy = jest.fn();
        const wrapper = mount(<GroupItems {...props} onItemSelection={methodSpy} />);

        wrapper.find('li').first().simulate('click');

        expect(methodSpy).toHaveBeenCalledWith(props.items[0]);
    });
});
