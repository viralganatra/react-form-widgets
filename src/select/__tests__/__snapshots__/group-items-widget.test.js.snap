exports[`GroupItemsWidget should render an array of items 1`] = `
<div>
  <mockConstructor
    label="test 1"
    onClick={[Function]} />
  <mockConstructor
    label="test 2"
    onClick={[Function]} />
</div>
`;

exports[`GroupItemsWidget should validate the prop types 1`] = `
"Warning: Failed prop type: The prop \`items[0].label\` is marked as required in \`SelectMultichoiceGroup\`, but its value is \`undefined\`.
    in SelectMultichoiceGroup"
`;

exports[`GroupItemsWidget should validate the prop types 2`] = `
"Warning: Failed prop type: Invalid prop \`onItemSelection\` of type \`number\` supplied to \`SelectMultichoiceGroup\`, expected \`function\`.
    in SelectMultichoiceGroup"
`;
